src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -320,56 +331,65 @@ public void renameSheet(String name, String newName) {
     }
 
     public void cloneSheet(int index, String newName) {
-        cloneSheet(workbook.getSheetName(index), newName);
+        Sheet sheet = workbook.cloneSheet(index);
+        String originalName = workbook.getSheetName(index);
+        workbook.setSheetName(workbook.getSheetIndex(sheet), newName);
+        //Copy names (named ranges) that are scoped to the original sheet, adapting the formula and scope.
+        List<Name> originalNamedRanges = workbook.getAllNames().stream().filter(name -> name.getSheetIndex() == index).collect(Collectors.toList());
+        // we have to collect the original names *then* add the new names, otherwise we get concurrency issues creating names while iterating over them
+        originalNamedRanges.forEach(namedRange ->
+                createName(namedRange.getNameName(), newName, namedRange.getRefersToFormula().replace(originalName, newName), false));
```

Comment:
@spoltier, I spotted this remaining inconsistency in terms of scope vs. reference sheet. I guess we can also drop `originalName` then.

```suggestion
        // Copy names (named ranges) that are scoped to the original sheet, adapting the scope.
        List<Name> originalNamedRanges = workbook.getAllNames().stream().filter(name -> name.getSheetIndex() == index).collect(Collectors.toList());
        // we have to collect the original names *then* add the new names, otherwise we get concurrency issues creating names while iterating over them
        originalNamedRanges.forEach(namedRange ->
                createName(namedRange.getNameName(), newName, namedRange.getRefersToFormula(), false));
```